packages/next/src/build/webpack-config.ts

```diff
@@ -1574,7 +1574,7 @@ export default async function getBaseWebpackConfig(
       (isEsm && isAppLayer)
 
     if (/node_modules[/\\].*\.[mc]?js$/.test(res)) {
-      if (layer === WEBPACK_LAYERS.server || layer === WEBPACK_LAYERS.action) {
+      if (layer && WEBPACK_LAYERS.GROUP.server.includes(layer)) {
```

Comment:
```suggestion
      if (WEBPACK_LAYERS.GROUP.server.includes(layer)) {
```

Reply:
TS requires that, we either check layer or use `layer!`, thought we want to avoid the `!` operator